### PR TITLE
[Impeller] fix incorrect origins for mesh gradient computation.

### PR DESCRIPTION
**Pull request by jonahwilliams**
Fixes https://github.com/flutter/flutter/issues/153964

Changing the origin of the rect used to render a shader could break shaders that expect to render at particular coordinates based on the input vertices. The snapshot functionality correctly handles translating a texture, so the translation was never necessary to begin with.

### CODE REVIEW

File: impeller/fixtures/runtime_stage_position.frag [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+// found in the LICENSE file.
+
+#include <flutter/runtime_effect.glsl>
+
+uniform vec4 ltrb;
+
+out vec4 frag_color;
+
+// Output solid red if frag position is within LTRB rectangle.
+void main() {
```
Comment (jonahwilliams) [inline impeller/fixtures/runtime_stage_position.frag:12]: This only outputs red if the rendered coordiantes match the expected coordinates.

File: impeller/display_list/aiks_dl_vertices_unittests.cc [hunk 2]
```diff
@@ -468,5 +468,54 @@ TEST_P(AiksTest, DrawVerticesTextureCoordinatesWithFragmentShader) {
   ASSERT_TRUE(OpenPlaygroundHere(builder.Build()));
 }
 
+// The vertices should form a solid red rectangle with no gaps.
+// The blue rectangle drawn under them should not be visible.
+TEST_P(AiksTest,
+       DrawVerticesTextureCoordinatesWithFragmentShaderNonZeroOrigin) {
+  std::vector<SkPoint> positions_lt = {
```
Comment (gaaclarke) [inline impeller/display_list/aiks_dl_vertices_unittests.cc:475]: No abbreviations ("lt").

File: impeller/fixtures/runtime_stage_position.frag [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+  if (FlutterFragCoord().x >= ltrb.x && FlutterFragCoord().x <= ltrb.z &&
+      FlutterFragCoord().y >= ltrb.y && FlutterFragCoord().y <= ltrb.w) {
+    frag_color = vec4(1.0, 0.0, 0.0, 1.0);
```
Comment (gaaclarke) [inline impeller/fixtures/runtime_stage_position.frag:15]: Wait, what is frag_color if it's outside of the box?  Shouldn't that be explicit here?

File: impeller/display_list/aiks_dl_vertices_unittests.cc [hunk 4]
```diff
@@ -468,5 +468,54 @@ TEST_P(AiksTest, DrawVerticesTextureCoordinatesWithFragmentShader) {
...
+  auto color_source = flutter::DlColorSource::MakeRuntimeEffect(
+      runtime_effect, {}, uniform_data);
+
+  paint.setColorSource(color_source);
+
+  builder.Scale(GetContentScale().x, GetContentScale().y);
+  builder.DrawRect(SkRect::MakeLTRB(200, 200, 250, 250), rect_paint);
+  builder.DrawVertices(vertices_lt, flutter::DlBlendMode::kSrcOver, paint);
+
+  ASSERT_TRUE(OpenPlaygroundHere(builder.Build()));
```
Comment (gaaclarke) [inline impeller/display_list/aiks_dl_vertices_unittests.cc:517]: Heads up you are going to need to add the included golden file to the list.

**Review by gaaclarke — COMMENTED** [review]
lgtm, just one question about uninitialized frag_color.

File: impeller/display_list/aiks_dl_vertices_unittests.cc (hunk 4 above)
Comment (jonahwilliams) [reply impeller/display_list/aiks_dl_vertices_unittests.cc:517]: I usually just wait for the check to fail first and then copy the string.

Comment (jonahwilliams) [reply impeller/display_list/aiks_dl_vertices_unittests.cc:517]: done

File: impeller/fixtures/runtime_stage_position.frag (hunk 3 above)
Comment (jonahwilliams) [reply impeller/fixtures/runtime_stage_position.frag:15]: updated

File: impeller/display_list/aiks_dl_vertices_unittests.cc (hunk 2 above)
Comment (jonahwilliams) [reply impeller/display_list/aiks_dl_vertices_unittests.cc:475]: done

**Review by gaaclarke — APPROVED** [review]
lgtm assuming the goldens look good to jonah

File: impeller/fixtures/runtime_stage_position.frag (hunk 3 above)
Comment (gaaclarke) [reply impeller/fixtures/runtime_stage_position.frag:15]: Considering we are cross compiling the shaders it doesn't hurt, thanks.

**Pull request merged**